indexing-service/src/main/java/io/druid/indexing/common/actions/TaskLockCriticalState.java

```diff
@@ -0,0 +1,18 @@
+package io.druid.indexing.common.actions;
+
+public enum TaskLockCriticalState
+{
+  UPGRADE {
+    public final boolean getExpectedState() {
+      return true;
+    }
+  },
+  DOWNGRADE {
+    public final boolean getExpectedState() {
+      return false;
+    }
+  };
+
+  public abstract boolean getExpectedState();
```

Comment:
you can have a variable "boolean expecteStage" , then overriding wouldn't be needed in all enum values.